### PR TITLE
Binder r

### PR DESCRIPTION
**Pull request by jburel**
This PR 
* adds the infrastructure to run the notebook using BinderHub
* links to run the notebook (when merged) on EBI BinderHub
* adds notebook to calculate the IC50 per compound

Few improvements will done afterwards but we need this in place before next week talk

cc @francesw @pwalczysko

### CODE REVIEW

**Comment by jburel** [comment]
https://github.com/jburel/idr0094-ellinger-sarscov2/actions/runs/747940577 is green

**Comment by pwalczysko** [comment]
Failing to build this locally with ``repo2docker``. How to test ?

**Comment by pwalczysko** [comment]
Deleted the comment, sorry, build in repo2docker locally as well as on mybinder in progress

File: notebooks/idr0094-r.ipynb [hunk 1]
```diff
@@ -0,0 +1,1441 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "names(df)[24] <- \"ConcX\"\n",
+    "names(df)[25] <- \"RespY\"\n",
+    "names(df)[13] <- \"CN\"\n",
+    "names(df)[19] <- \"InChIKey\""
```
Comment (pwalczysko) [inline notebooks/idr0094-r.ipynb]: Not sure - this seems like a non-intuitive and hard-coded to me. Is this going to repeat for any study ? I suppose there would be an option to first get the names suggestion from the dataframe itself and then, (with human interaction or without it) to rename the columns accordingly ?

File: notebooks/idr0094-r.ipynb [hunk 2]
```diff
@@ -0,0 +1,1441 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "names(df)[24] <- \"ConcX\"\n",
+    "names(df)[25] <- \"RespY\"\n",
```
Comment (pwalczysko) [inline notebooks/idr0094-r.ipynb]: Why is this column renamed to "RespY" ? According to what I see from the dataframe on the notebook, the column 25 is the one with the "Percentage inhibition DPC" value ? This means these numbers will end up on the Y axis ?
What is RespY an abbreviation of ? Maybe I do not fully understand the flow of data here.


Screenshot 2021-04-14 at 15 05 07

File: notebooks/idr0094-r.ipynb [hunk 3]
```diff
@@ -0,0 +1,1441 @@
...
+   "source": [
+    "ic <- vector()\n",
+    "cn <- vector()\n",
+    "for(i in 1:nrow(compounds)) {\n",
+    "    c <- compounds[i,]\n",
+    "    if (c != \"\") {\n",
+    "      data <- filter_data(df, c)\n",
+    "      results <- calculate_IC50(data)\n",
+    "      ic <- c(ic, results$ic50)\n",
+    "      cn <- c(cn, c) \n",
```
Comment (pwalczysko) [inline notebooks/idr0094-r.ipynb]: the code is fine as is I suppose, but for the printout of the table in the notebook, a bit more verbose variant of the column headers would be appropriate imho

Screenshot 2021-04-14 at 15 06 44

File: notebooks/idr0094-r.ipynb [hunk 4]
```diff
@@ -0,0 +1,1441 @@
...
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Export result as CSV\n",
+    "\n",
+    "Write the calculated IC50 result into a CSV file. The CSV file is first created in the ``home`` directory. The CSV file can then be downloaded to your local machine. To download it, click ``File > Open``, select the CSV file and open it, then click ``File > Download``."
```
Comment (pwalczysko) [inline notebooks/idr0094-r.ipynb]:
```suggestion
    "Write the calculated IC50 result into a CSV file. The CSV file is first created in the ``home`` directory. The CSV file can then be downloaded to your local machine. To download it, click ``File > Open``, navigate to the ``home`` directory (click onto the folder icon in top-left corner), select the CSV file and open it, then click ``File > Download``."
```

Comment (pwalczysko) [reply notebooks/idr0094-r.ipynb]: For the results in the form of CSV, suggest we need to 

1. export better column header names, as commented on https://github.com/IDR/idr0094-ellinger-sarscov2/pull/1/files#r613280654
2. more importantly, there are no units for the ``ic`` column, which makes it less useful

**Comment by pwalczysko** [comment]
Couple of comments, otherwise works nicely (tested locally with repo2docker). MyBinder was not successful in building this, as indicated by @jburel

File: notebooks/idr0094-r.ipynb (hunk 1 above)
Comment (jburel) [reply notebooks/idr0094-r.ipynb]: The problem is b/c we have column with space which implies that access is painful and it also prevents the filtering when loading the data from server.
The rename is for convenience and it simplifies access.

**Comment by jburel** [comment]
@pwalczysko:
*  I have adjusted the hard-coded names that are used to simplify access to the columns
* I have added names to the result table

**Comment by pwalczysko** [comment]
I am not sure about the values here for remdesivir to start with.
So, when studying the output of the CSV produced by the notebook, then I am getting  (the units are ``M``, this means Molar)

Screenshot 2021-04-14 at 17 00 53

which translates into 489773 microM (four hundred thousand microM, seems to me way too high)

And if I let the notebook to print out the value as follows

```
values <- calculate_IC50(data)
IC50 <- values$ic50
print (IC50)
```

I get 2.381737 microMolar

But the paper states clearly
``The IC50 of the RNA polymerase inhibitor remdesivir (0.76 ± 0.18 µM) ``

This seems to me like 3 wildly different numbers.

File: notebooks/idr0094-r.ipynb (hunk 1 above)
Comment (pwalczysko) [reply notebooks/idr0094-r.ipynb]: Yes, understand that the rename is necessary. Just wondering how did you identify the correct columns to rename in first place ? There must have been a step which was reading the columns and identifying the headers and then renaming these - was this purely manual ? Like find in the dataframe the column named "Concentration <space> something" and then noting its number (24) ? I think it is not too bad, but if there is an easy way to do it semi-automatically, we might expose it.

**Comment by pwalczysko** [comment]
I think the notebook is working nicely, but a bit worried about the values returned https://github.com/IDR/idr0094-ellinger-sarscov2/pull/1#issuecomment-819635487

**Comment by jburel** [comment]
sorry I fixed the unit.

**Comment by jburel** [comment]
I don't know where you got `` 2.381737 microMolar`` Are you sure it is for ``Remdesivir``?
This is the value for Remdesivir
Screenshot 2021-04-15 at 09 48 56

**Comment by jburel** [comment]
@pwalczysko in order to integrate with IDR and not hard-code value if we want to run it on screen A. The screenID is passed via the URL. When you try locally could you try URL like ``http://127.0.0.1:60982/notebooks/notebooks/idr0094-r.ipynb?screenId=2603`` (port will be different for you)

**Comment by pwalczysko** [comment]
Tested locally via repo2docker.

All good here and makes sense, units, the graph and the renaming too.
Ready to merge fmpov

**Comment by jburel** [comment]
Screenshot 2021-04-15 at 13 19 04

**Comment by pwalczysko** [comment]
Worked now on binderhub too.

**Review by pwalczysko — APPROVED** [review]

**Comment by jburel** [comment]
Renamed notebook and updated link in the readme
Merging.
I might review the IC50 in a follow-up PR

**Pull request merged**